rustls/src/client/handy.rs

```diff
@@ -241,6 +242,44 @@ impl client::ResolvesClientCert for AlwaysResolvesClientCert {
     }
 }
 
+#[derive(Debug)]
+pub(super) struct AlwaysResolvesClientRawPublicKeys(Arc<sign::CertifiedKey>);
+
+impl AlwaysResolvesClientRawPublicKeys {
+    pub(super) fn new(private_key: Arc<dyn sign::SigningKey>) -> Result<Self, Error> {
+        let public_key_as_cert = CertificateDer::from(
+            private_key
+                .public_key()
+                .unwrap()
+                .into_owned()
+                .as_ref()
+                .to_vec(),
+        );
+        Ok(Self(Arc::new(sign::CertifiedKey::new(
+            CertificateChain(vec![public_key_as_cert]).0,
+            private_key,
+        ))))
+    }
+}
+
+impl client::ResolvesClientCert for AlwaysResolvesClientRawPublicKeys {
+    fn resolve(
+        &self,
+        _root_hint_subjects: &[&[u8]],
+        _sigschemes: &[SignatureScheme],
+    ) -> Option<Arc<sign::CertifiedKey>> {
+        Some(Arc::clone(&self.0))
+    }
+
+    fn only_raw_public_keys(&self) -> bool {
+        true
+    }
+
+    fn has_certs(&self) -> bool {
+        true
+    }
```

Comment:
This feels a bit odd to me. There's not really a certificate available right?

Renaming the `fn` to something more about the resolver being ready to present an identity seems appealing but would be a breaking change. Maybe a compromise would be a comment here explaining some of the nuance?

Reply:
Added a comment that mentions the nuance